raco/backends/myria/myria.py

```diff
@@ -1785,6 +1807,8 @@ def opt_rules(self, **kwargs):
             rules.push_select,
             rules.push_project,
             rules.push_apply,
+            [PushSelectThroughShuffle()],
+            rules.push_select,
```

Comment:
That is intentional. Since the existing `PushSelects` rule cannot push selects below a shuffle, it has no opportunity to perform optimizations below the shuffle (e.g., pushing the select down as far as possible, merging with other selects). Therefore I need to run the rule (actually the set of rules in `raco.rules.push_select`) again to ensure that selects just pushed below a shuffle are fully optimized. There are several other examples of this in RACO, e.g., the definition of `push_apply` in `raco/raco/rules.py`:

```
push_apply = [
    # These really ought to be run until convergence.
    # For now, run twice and finish with PushApply.
    PushApply(),
    RemoveUnusedColumns(),
    PushApply(),
    RemoveUnusedColumns(),
    PushApply(),
    RemoveNoOpApply(),
]
```